src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,51 @@
 
 class Animal {
   // write your code here
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  static removeAnimal(animal) {
+    const animalIndex = this.alive.findIndex(el => el === animal);
+
+    this.alive.splice(animalIndex, 1);
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.addAnimal(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(target) {
+    if (target instanceof Herbivore) {
+      if (target.hidden === false) {
+        target.health -= 50;
+
+        if (target.health <= 0) {
+          Animal.removeAnimal(target);
```

Comment:
I did that because the Animal object can be created with custom health, and if not then it will be 100. So if you make it, for example, 49, after bite() it will be less than 0.